sklearn/feature_selection/_sequential.py

```diff
@@ -172,21 +184,32 @@ def fit(self, X, y):
         # - that we have already *excluded* if we do backward selection
         current_mask = np.zeros(shape=n_features, dtype=bool)
         n_iterations = (
-            self.n_features_to_select_ if self.direction == 'forward'
+            self.n_features_to_select_
+            if (self.n_features_to_select == 'auto') or (
+                     self.direction == 'forward')
             else n_features - self.n_features_to_select_
         )
+
+        old_score = -np.inf
         for _ in range(n_iterations):
-            new_feature_idx = self._get_best_new_feature(cloned_estimator, X,
-                                                         y, current_mask)
+            new_feature_idx, new_score = self._get_best_new_feature_score(
+                cloned_estimator, X, y, current_mask)
+            if self.n_features_to_select == 'auto' and (
+                    new_score - old_score) < self.tol:
```

Comment:
We'll probably need to check that self.tol is not None first

Reply:
fix it on https://github.com/scikit-learn/scikit-learn/pull/20145/commits/9b11fbbcfb1b443063a25940f4f8d3f5a4c6951e